### PR TITLE
(#1037) Add equals and hashCode to IterableEnvelope

### PR DESCRIPTION
**Pull request by Umbrah**
PR for #1037 
I added equals() and hashCode() to IterableEnvelope.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1101](https://github.com/yegor256/cactoos/issues/1101) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1101](https://github.com/yegor256/cactoos/issues/1101) is assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/C63314D6Z/35389175-2787-431a-a252-f0160f011065); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by victornoel** [comment]
@Umbrah can you take the habit of putting the issue number in the description when you open a PR? It's always a pain to find back the original issue when it's only in the title of the PR :)

**Comment by Umbrah** [comment]
@victornoel Sure, my fault :) Updated this one.

File: src/test/java/org/cactoos/iterable/IterableEnvelopeTest.java [hunk 1]
```diff
@@ -49,4 +52,75 @@ public void returnsIteratorWithUnsupportedRemove() {
...
+            new IsEqual<>(new IterableOf<>())
+        ).affirm();
+    }
+
+    @Test
+    public void differentHashCode() {
+        final IterableOf<Integer> first = new IterableOf<>(1, 2);
+        final IterableOf<Integer> second = new IterableOf<>(2, 1);
+        new Assertion<>(
+            "hashCode is equal for Iterables with different content",
```
Comment (victornoel) [inline src/test/java/org/cactoos/iterable/IterableEnvelopeTest.java]: @Umbrah please rephrase this with the "must xxxx" formulation

File: src/test/java/org/cactoos/iterable/IterableEnvelopeTest.java [hunk 2]
```diff
@@ -49,4 +52,75 @@ public void returnsIteratorWithUnsupportedRemove() {
...
+            first::hashCode,
+            new IsNot<>(new IsEqual<>(second.hashCode()))
+        ).affirm();
+    }
+
+    @Test
+    public void equalHashCode() {
+        final IterableOf<Integer> iterable = new IterableOf<>(1, 2);
+        new Assertion<>(
+            "hashCode is different for Iterables with equal content",
```
Comment (victornoel) [inline src/test/java/org/cactoos/iterable/IterableEnvelopeTest.java]: @Umbrah please rephrase this with the "must xxxx" formulation

**Comment by Umbrah** [comment]
@victornoel Please check my update

**Comment by victornoel** [comment]
@llorllale it's good to merge

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @victornoel[/z](https://www.0crat.com/u/victornoel), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1101](https://github.com/yegor256/cactoos/issues/1101) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @victornoel[/z](https://www.0crat.com/u/victornoel)

**Pull request merged**